extension/controller/experiments/TranslationBar/api.js

```diff
@@ -115,9 +120,12 @@ const translationNotificationManagers = new Map();
                 if (!windowsWithCustomElement.has(chromeWin)) {
                   windowsWithCustomElement.add(chromeWin);
                   chromeWin.TRANSLATION_NOTIFICATION_ELEMENT_ID = TRANSLATION_NOTIFICATION_ELEMENT_ID;
-                  Services.scriptloader.loadSubScript(
+                  Services.scriptloader.loadSubScriptWithOptions(
                     context.extension.getURL("view/js/translation-notification-fxtranslations.js"),
-                    chromeWin
+                    {
+                      target: chromeWin,
+                      ignoreCache
```

Comment:
This part was refactored by @Rob--W when we were having issues with the translationbar not being displayed intermittently, but what I can confirm based on observations made using the debugger you showed me yesterday, is that this condition [1]
returns true only the first time that the method `show` is called, i.e, only the first time a translation bar is displayed. All the subsequent ones return false, except when a new window is open, of course.


[1]

https://github.com/mozilla/firefox-translations/blob/dc549ac46858a291f8bc873ef2b644c0827e339d/extension/controller/experiments/TranslationBar/api.js#L115

Reply:
@andrenatal yep this is going to be executed once for every new chrome window, what I meant is that in theory this may need to pass ignoreCache only one time to refresh a subscript that may have been cached by a previous version, at the moment ignoreCache will stay true for each new chrome window opened while the extension running just got updated, then in the next browser or addon startup ignoreCache will stay false and the script will be cached.

Yesterday I briefly touched base with @mixedpuppy about this corner case and we were in agreement that keeping it as is may be the safest option at the moment (and so to defer doing something about it to either a separate followup or if that turns out to have some impact on user perceived performance).

May still be worth capturing this corner case and related details collected about it in an issue to keep that in mind.

Reply:
done